src/Storefront/Controller/AddressController.php

```diff
@@ -208,47 +226,191 @@ public function addressBook(Request $request, RequestDataBag $dataBag, SalesChan
         return $response;
     }
 
+    #[Route(path: '/account/address/delete/{addressId}', name: 'frontend.account.address.delete', options: ['seo' => false], defaults: ['XmlHttpRequest' => true, '_loginRequired' => true], methods: ['POST'])]
+    public function deleteAddress(string $addressId, Request $request, SalesChannelContext $context, CustomerEntity $customer): Response
+    {
+        $success = true;
+
+        if (!$addressId) {
+            throw RoutingException::missingRequestParameter('addressId');
+        }
+
+        try {
+            $this->deleteAddressRoute->delete($addressId, $context, $customer);
+        } catch (InvalidUuidException|AddressNotFoundException|CannotDeleteDefaultAddressException|CustomerException $e) {
+            dump($e->getMessage());
+            $success = false;
+        }
+
+        return new RedirectResponse($this->generateUrl('frontend.account.address.page', ['addressDeleted' => $success]));
```

Comment:
is `addressDeleted` used for something else? how about adding a flash about the success/failure of the deletion like in your other methods?